main.go

```diff
@@ -228,8 +227,16 @@ func main() { //nolint:funlen
 		setupLog.Error(err, "unable to update from legacy operator version")
 	}
 
-	if err = upgrade.CleanupExistingResource(setupClient, platform); err != nil {
-		setupLog.Error(err, "unable to perform cleanup")
+	var cleanExistingResourceFunc manager.RunnableFunc = func(ctx context.Context) error {
+		if err = upgrade.CleanupExistingResource(ctx, setupClient, platform, dscApplicationsNamespace, dscMonitoringNamespace); err != nil {
+			setupLog.Error(err, "unable to perform cleanup")
+		}
+		return err
+	}
+	err = mgr.Add(cleanExistingResourceFunc)
+	if err != nil {
+		setupLog.Error(err, "error remove deprecated resources from previous version")
+		os.Exit(1)
```

Comment:
Should we continue with error log if there is error in resource removal i.e remove `os.Exit(1)`